model-optimizer/extensions/front/tf/KerasRNNReplacer.py

```diff
@@ -0,0 +1,237 @@
+"""
+ Copyright (C) 2017-2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from extensions.front.tf.WhileNormalize import WhileNormalize
+from extensions.ops.loop import Loop
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.front.tf.graph_utils import create_op_with_const_inputs
+from mo.graph.graph import Graph, Node, rename_nodes
+from mo.middle.pattern_match import find_pattern_matches, inverse_dict
+from mo.ops.const import Const
+from mo.ops.squeeze import Squeeze
+from mo.ops.unsqueeze import Unsqueeze
+
+
+def find_intergraph_match(graph: Graph, ext_match: dict, reserve_port: int, unstack_port: int, stack_port: int):
```

Comment:
The transformation handles LSTM and GRU only. But there are much more other types of RNN operations: https://keras.io/api/layers/recurrent_layers/. For example, Bidirectional https://keras.io/api/layers/recurrent_layers/bidirectional/ are the important ones because we support them from TF 1.x.

Reply:
Now, we manage to cover LSTM, GRU, and SimpleRNN. Other layers will be covered in separate activity as we agreed.